contracts/cw-names-registry/src/contract.rs

```diff
@@ -0,0 +1,256 @@
+    let config = CONFIG.load(deps.storage)?;
+
+    // We only take payments from our specified token
+    if info.sender != config.payment_token_address {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // TODO: Is there a way we can verify this is a DAO
+    let sender = wrapped.sender;
+    let amount = wrapped.amount;
+    let msg: ReceiveMsg = from_binary(&wrapped.msg)?;
+
+    match msg {
+        ReceiveMsg::Register { name } => register_name(deps, env, sender, amount, name),
+    }
+}
+
+pub fn register_name(
+    deps: DepsMut,
+    _env: Env,
+    sender: String,
+    amount: Uint128,
+    name: String,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    if amount < config.payment_amount {
+        return Err(ContractError::InsufficientFunds {});
+    }
+
+    // this is the DAO
+    let sender = deps.api.addr_validate(&sender)?;
+
+    if RESERVED_NAMES.has(deps.storage, name.clone()) {
+        return Err(ContractError::NameReserved {});
+    }
+
+    if NAME_TO_DAO.has(deps.storage, name.clone()) {
+        return Err(ContractError::NameAlreadyTaken {});
+    }
+
+    if DAO_TO_NAME.has(deps.storage, sender.clone()) {
+        return Err(ContractError::NameAlreadyRegistered {});
+    }
+
+    NAME_TO_DAO.save(deps.storage, name.clone(), &sender)?;
+    DAO_TO_NAME.save(deps.storage, sender, &name)?;
+
+    Ok(Response::new())
+}
+
+pub fn execute_update_config(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    new_payment_token_address: Option<String>,
+    new_admin: Option<String>,
+    new_payment_amount: Option<Uint128>,
+) -> Result<Response, ContractError> {
+    let mut config = CONFIG.load(deps.storage)?;
+
+    if info.sender != config.admin {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    let new_payment_token_address =
+        new_payment_token_address.unwrap_or_else(|| config.payment_token_address.to_string());
+    let payment_amount = new_payment_amount.unwrap_or(config.payment_amount);
+    let new_admin = new_admin.unwrap_or_else(|| config.admin.to_string());
+
+    // Validate addresses
+    let admin = deps.api.addr_validate(&new_admin)?;
+    let payment_token_address = deps.api.addr_validate(&new_payment_token_address)?;
+
+    config.payment_amount = payment_amount;
+    config.admin = admin;
+    config.payment_token_address = payment_token_address;
+
+    CONFIG.save(deps.storage, &config)?;
+
+    Ok(Response::new().add_attribute("action", "update_config"))
+}
+
+pub fn execute_revoke(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    name: String,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    let dao = NAME_TO_DAO.load(deps.storage, name.clone())?;
+
+    // Only name owner and overall name admin can revoke
+    if info.sender != dao && info.sender != config.admin {
+        return Err(ContractError::Unauthorized {});
+    }
```

Comment:
I might be dumb, but I'm confused here. Shouldn't this be `info.sender != dao || info.sender != config.admin`? You can revoke the name if you are the Naming DAO (admin) *or* if you're the owner of the name. Right? Am I missing something?

Nitpick: instead of "overall name admin" in the comment, I'd probably say something like, "admin of the contract (which we anticipate to be the Name Registry DAO)". Making our expectations explicit is always good.

Reply:
yep you're right good spot